sdk/cosmosdb/cosmos/src/queryExecutionContext/ContinuationTokenManager/ContinuationTokenManagerFactory.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { BaseContinuationTokenManager } from "./BaseContinuationTokenManager.js";
+import { ParallelQueryContinuationTokenManager } from "./ParallelQueryContinuationTokenManager.js";
+import { OrderByQueryContinuationTokenManager } from "./OrderByQueryContinuationTokenManager.js";
+
+/**
+ * Factory for creating appropriate continuation token managers based on query type.
+ * @internal
+ */
+export const ContinuationTokenManagerFactory = {
+  /**
+   * Creates the appropriate continuation token manager based on query type.
+   * @param collectionLink - Collection link/RID
+   * @param initialContinuationToken - Optional initial continuation token
+   * @param isOrderByQuery - Whether this is an ORDER BY query
+   * @param isUnsupportedQueryType - Whether this query type doesn't support continuation tokens
+   * @returns Appropriate continuation token manager instance
+   * @internal
+   */
+  create(
+    collectionLink: string,
+    initialContinuationToken?: string,
+    isOrderByQuery: boolean = false,
+    isUnsupportedQueryType: boolean = false,
+  ): BaseContinuationTokenManager {
+    
+    if (isOrderByQuery) {
```

Comment:
Why do we even need to create continuation token manager when we know that it is not supported for a query type.